File: app/src/main/java/ch/zuehlke/sbb/reddit/features/login/LoginContract.java
package ch.zuehlke.sbb.reddit.features.login;

import ch.zuehlke.sbb.reddit.BasePresenter;
import ch.zuehlke.sbb.reddit.BaseView;

/**
 * Created by chsc on 08.11.17.
 */

public interface LoginContract {

    interface View extends BaseView<Presenter>{

        void setLoadingIndicator(boolean isActive);

        void showInvalidUsername();

        void showInvalidPassword();

        void showRedditNews();
    }

    interface Presenter extends BasePresenter{

        void login(String userEmail,String password);
    }
}

File: app/src/main/java/ch/zuehlke/sbb/reddit/features/login/LoginContract.kt
package ch.zuehlke.sbb.reddit.features.login

import ch.zuehlke.sbb.reddit.BasePresenter
import ch.zuehlke.sbb.reddit.BaseView

/**
 * Created by chsc on 08.11.17.
 */

interface LoginContract {

    interface View : BaseView<Presenter> {

        fun setLoadingIndicator(isActive: Boolean)

        fun showInvalidUsername()

        fun showInvalidPassword()

        fun showRedditNews()
    }

    interface Presenter : BasePresenter {

        fun login(userEmail: String, password: String)
    }
}
